packages/contracts/contracts/rewards/SubgraphAvailabilityManager.sol

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.7.6;
+
+import { SafeMathUpgradeable } from "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+
+import { Governed } from "../governance/Governed.sol";
+import { IRewardsManager } from "../rewards/IRewardsManager.sol";
+
+/**
+ * @title Subgraph Availability Manager
+ * @dev Manages the availability of subgraphs by allowing oracles to vote on whether
+ * a subgraph should be denied rewards or not. When enough oracles have voted to deny or
+ * allow rewards for a subgraph, it calls the RewardsManager Contract to set the correct
+ * state. The number of oracles and the execution threshold are set at deployment time.
+ * Only governance can set the oracles.
+ */
+contract SubgraphAvailabilityManager is Governed {
+    using SafeMathUpgradeable for uint256;
+
+    // -- Immutable --
+
+    /// @notice Number of oracles
+    uint256 public constant NUM_ORACLES = 5;
+
+    /// @notice Number of votes required to execute a deny or allow call to the RewardsManager
+    uint256 public immutable executionThreshold;
+
+    /// @dev Address of the RewardsManager contract
+    IRewardsManager private immutable rewardsManager;
+
+    // -- State --
+
+    /// @dev Nonce for generating votes on subgraph deployment IDs.
+    /// Increased whenever oracles or voteTimeLimit change, to invalidate old votes.
+    uint256 private currentNonce;
```

Comment:
Could we make this public just for convenience? it could be useful to query this easily if we want to manually inspect votes

Reply:
Yup, seems fair.